crates/uv/src/target.rs

```diff
@@ -0,0 +1,142 @@
+use pep508_rs::MarkerEnvironment;
+use platform_tags::{Arch, Os, Platform};
+
+/// The supported target triples. Each triple consists of an architecture, vendor, and operating
+/// system.
+///
+/// See: <https://doc.rust-lang.org/nightly/rustc/platform-support.html>
+#[derive(Debug, Clone, Copy, Eq, PartialEq, clap::ValueEnum)]
+pub(crate) enum TargetTriple {
+    #[value(name = "x86_64-pc-windows-msvc")]
+    X8664PcWindowsMsvc,
+    #[value(name = "x86_64-unknown-linux-gnu")]
+    X8664UnknownLinuxGnu,
+    #[value(name = "x86_64-apple-darwin")]
+    X8664AppleDarwin,
+    #[value(name = "aarch64-apple-darwin")]
+    Aarch64AppleDarwin,
+    #[value(name = "aarch64-unknown-linux-gnu")]
+    Aarch64UnknownLinuxGnu,
+    #[value(name = "aarch64-unknown-linux-musl")]
+    Aarch64UnknownLinuxMusl,
+    #[value(name = "x86_64-unknown-linux-musl")]
+    X8664UnknownLinuxMusl,
+}
+
+impl TargetTriple {
+    /// Return the [`Platform`] for the target.
+    pub(crate) fn platform(self) -> Platform {
+        match self {
+            Self::X8664PcWindowsMsvc => Platform::new(Os::Windows, Arch::X86_64),
+            Self::X8664UnknownLinuxGnu => Platform::new(
+                Os::Manylinux {
+                    major: 2,
+                    minor: 17,
```

Comment:
All of these values are based on the Tier 1 and Tier 2 definitions in the Rust project.

Reply:
I think we should add variants for the other manylinux versions. (By "failed", you mean, "didn't download those wheels"? Or, how did it fail?)

Reply:
e.g. [open3d](https://pypi.org/project/open3d/#files) uses `glibc` 2.27 and it fails like this:

```bash
❯ uv pip compile requirements.in --python-platform=linux
  x No solution found when resolving dependencies:
  `-> Because open3d==0.18.0 is unusable because no wheels are available with a matching Python implementation
   and you require open3d==0.18.0, we can conclude that the requirements are unsatisfiable.
```

Reply:
Thanks.

Reply:
Filed here: https://github.com/astral-sh/uv/issues/3222. I have an idea for the format.

Reply:
In that case, should it be able to pick out the universal wheels?

Reply:
Hmm good question, I think so?, because it would match the platform.

Reply:
I'll look into that separately.

Reply:
Yes, with `failed` I meant [PEP 600](https://peps.python.org/pep-0600/) `manylinux_x_y` packages not being selected during `uv pip compile` due to a glibc version mismatch. Of course this behavior is expected, though an 11 year old version does effectively create constraints.

For us a few more allowed options for the `python-platform` string would probably already work, thx :+1: